src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                findBookById = new Book(bookId, title, price);
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
+        }
+        return Optional.ofNullable(findBookById);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> foundBooks = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.connectToDatabase();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(FIND_ALL_BOOK_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                foundBooks.add(new Book(bookId, title, price));
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
+        }
+        return foundBooks;
+    }
+
+    @Override
+    public Book update(Book book) {
+        Book updatedBook = null;
+        try (Connection connection = ConnectionUtil.connectToDatabase();
+                PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_BOOK_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(2, book.getTitle());
+            preparedStatement.setBigDecimal(3, book.getPrice());
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                updatedBook = new Book(bookId, title, price);
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
+        }
+        return updatedBook;
+    }
+
+    @Override
+    public boolean deleteById(int id) {
+        boolean isDeleted;
+        try (Connection connection = ConnectionUtil.connectToDatabase();
+                PreparedStatement preparedStatement = connection.prepareStatement(DELETE_BOOK_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, id);
+            isDeleted = preparedStatement.execute();
+        } catch (SQLException throwables) {
+            throw new RuntimeException(throwables);
```

Comment:
Use a more informative message for the RuntimeException, such as including the action that was attempted when the exception occurred.